miden/src/lib.rs

```diff
@@ -5,8 +5,9 @@
 // ================================================================================================
 
 pub use assembly::{
-    ast::{ModuleAst, ProgramAst},
-    Assembler, AssemblyError, ParsingError,
+    self,
+    ast::{Module, ModuleKind},
+    diagnostics, Assembler, AssemblyError,
 };
```

Comment:
That's what I would be inclined to do, but I guess I didn't want to remove some of the "major" types from the crate root, so as to avoid more downstream changes, but honestly it's not a big deal to fix those up, so if you'd prefer to just export the `assembly` module and drop the top-level re-export of things like `Module` and `Assembler`, that's fine by me!